edb/pgsql/resolver/expr.py

```diff
@@ -87,11 +100,38 @@ def resolve_ResTarget(
         res = []
         columns = []
         for table, column in col_res:
-            columns.append(column)
+            val = resolve_column_kind(table, column.kind, ctx=ctx)
+
+            # make sure name is not duplicated
+            # this behavior is technically different then Postgres, but EdgeDB
+            # protocol does not support duplicate names. And we doubt that
+            # anyone is depending on original behavior.
+            nam: str = column.name
```

Comment:
`nam` for `name` is a more aggressive abbreviation than I would normally do :P

Reply:
That's because variable `name` is already used down the function and python doesn't know about "block scopes". It does have the same type (str), but mypy was complaining about variable being re-defined.